### PR TITLE
Generalize save/load GUI

### PR DESCRIPTION
**Pull request by oOKexOo**
**When merged this pull request will:**
- Revise and generalize the save/load GUI for advanced compositions, mission SQF, _etc._

### CODE REVIEW

File: @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// safe conversion from string to array which prevents code injection
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf]: `/*`

File: @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+// makes use of parseSimpleArray
+// parseSimpleArray has a very strict format:
+// 1) no spaces allowed
+// 2) only double quotes for strings
+// The function takes care of 1) and 2)
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf]: `*/`

File: @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+params [["_string", "[]", [""]]];
+
+private _charArray = toArray _string;
+private _isInsideQuote = false;
+private _isApostrophQuote = false;
+for "_idx" from (count _charArray -1) to 0 step -1 do
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf]: Why not `_i`? To keep things standard.

File: @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+{
+	private _char = _charArray select _idx;
+	switch (true) do
+	{
+		// remove whitespace outside strings
+		case (_char in [0x09, 0x0A, 0x0D,0x20] && {!_isInsideQuote}):
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf]: `0x0D, 0x20`

File: @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+{
+	private _char = _charArray select _idx;
+	switch (true) do
+	{
+		// remove whitespace outside strings
+		case (_char in [0x09, 0x0A, 0x0D,0x20] && {!_isInsideQuote}): 
+		{
+			_charArray deleteAt _idx;
+		};
+		// handle apostroph
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_ares/DevTools/functions/fn_stringToArray.sqf]: Apostrophe

File: @AresModAchillesExpansion/addons/ui_f/dialogs/RscDialogLoadSave.hpp [hunk 6]
```diff
@@ -0,0 +1,240 @@
+class Achilles_rsc_loadSave
+{
+	idd = 133800;
+	movingEnable = true;
+	onLoad = "([""LOAD""] + _this) call Achilles_fnc_RscDisplayAttributes_loadSave";
+	onUnload = "([""UNLOAD""] + _this) call Achilles_fnc_RscDisplayAttributes_loadSave";
+	class controls 
+	{
+		// top
+		class Achilles_ctrl_addonIconBackground: IGUIBack
```
Comment (oOKexOo) [inline @AresModAchillesExpansion/addons/ui_f/dialogs/RscDialogLoadSave.hpp]: Can I review my own stuff?

**Pull request closed without merging**